docs/source/testing.rst

```diff
@@ -0,0 +1,98 @@
+.. _`tox`: https://tox.readthedocs.io/en/latest/
+.. _`nise`: https://github.com/project-koku/nise
+
+~~~~~~~~~~~~
+Testing Koku
+~~~~~~~~~~~~
+
+Unit testing
+============
+
+Unit testing for Koku is handled through `tox`_.
+
+Running the test suite requires a minimum of an accessible database and Koku API server.
+
+Unit testing using docker-compose
+---------------------------------
+
+Example:
+::
+    $ make docker-up
+    $ tox
+    $ make docker-down
+
+Unit testing using OpenShift
+---------------------------------
+
+The ``Makefile`` provides convenience commands for enabling port-forwarding. This is required for accessing a database pod hosted within an OpenShift environment.
+
+Example:
+::
+    $ make oc-create-all
+    $ make oc-forward-ports
+    $ tox
+    $ make stop-forwarding-ports
+
+Smoke testing
+=============
+
+Running the QE functional and smoke tests is achieved through the scripts provided in ``koku.git/testing``
+
+Requirements
+------------
+In order to run the test suite, there need to be clones of the following git repositories:
```

Comment:
Are these requirements needed for local OpenShift IQE testing?  When running with compose at least you don't have to have any of these installed or configured on your system since they are pre-built in the IQE container.

Reply:
The `e2e-deploy.sh` script leverages the `iqe` command to deploy a portion of the environment.